### PR TITLE
Setup automated testing for CUDA backend on GitLab CI

### LINKED ISSUE #19: Register GPU machine for Cuda builds through gitlab.com
Here is how you register a runner:
https://docs.gitlab.com/runner/register/

then we need to modify our `.gitlab-ci.yml` to use that runner on bare metal.

### PR DESCRIPTION
**Pull request by dalg24**
Close #19

### CODE REVIEW

**Comment by dalg24** [comment]
@junghans it did not trigger the build on GitLab :/

I will go ahead and push on a branch on the main repo to debug in-between but we need to figure out why the build was triggered for #78 but not here.

**Comment by dalg24** [comment]
@junghans The CI job does what we want (see [here](https://gitlab.com/ECP-CoPA/Cabana/-/jobs/174176722)).  Would you please look into the issue with triggering the CI builds from a fork?

**Comment by dalg24** [comment]
> @junghans The CI job does what we want (see [here](https://gitlab.com/ECP-CoPA/Cabana/-/jobs/174189866)). Would you please look into the issue with triggering the CI builds from a fork?
> 
> @sslattery Would you please make the AoSoA test pass? (It belong to another PR)

AoSoA test has been fixed in #80 that just got merged into master.  Will rebase on top of it but waiting on the build trigger to be looked at.

**Comment by junghans** [comment]
@dalg24 It seems that PR from forks only get triggered for existing PRs when gitlab mirror is setup, but not for newly created PRs after that, see: https://gitlab.com/gitlab-org/gitlab-ee/issues/5667#note_145669799.

There is a workaround by triggering a branch creatiing on gitlab.com for every PR from a member via a webhook.

**Comment by junghans** [comment]
`git fetch origin pull/79/head:pr_79 && git push gitlab pr_79 && git branch -D pr_79`

File: scripts/gitlab/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+FROM nvidia/cuda:10.0-devel-ubuntu18.04
+RUN apt-get update && apt-get install -y \
+        bc \
+        wget \
+    && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install CMake
+RUN export CMAKE_VERSION=3.13.4 && \
```
Comment (junghans) [inline scripts/gitlab/Dockerfile]: Could be just use Ubuntu's cmake?

Comment (dalg24) [reply scripts/gitlab/Dockerfile]: Yes if we stick to CUDA 10.0
But say you only have CUDA 9.2 on your machine, then you'd have to change the base image to `nvidia/cuda:9.2-devel-ubuntu16.04` (18.04 is not available) which comes with cmake-3.5.1 which is too old for Cabana.  I did not bother making the base image a build argument but it is intended as a customization point.

**Comment by dalg24** [comment]
> git fetch origin pull/79/head:pr_79 && git push gitlab pr_79 && git branch -D pr_79

That defeats the purpose if we do it by hand.  Do you know how to setup such a webhook?

**Comment by dalg24** [comment]
Closing until GitLab supports triggering the CI from PRs that have been opened from developers forks.

**Pull request closed without merging**